Core/include/Acts/Seeding/BinnedGroup.hpp

```diff
@@ -0,0 +1,104 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Seeding/BinnedGroupIterator.hpp"
+#include "Acts/Seeding/SpacePointGrid.hpp"
+#include "Acts/Utilities/GridBinFinder.hpp"
+#include "Acts/Utilities/GridIterator.hpp"
+#include "Acts/Utilities/Holders.hpp"
+
+#include <memory>
+#include <vector>
+
+namespace Acts {
+
+/// @class BinnedGroup
+/// @tparam grid_t Type of the grid the group owns
+///
+/// The assumption is that the grid has ownership of the space points and
+/// that the grid value_type (i.e. T in Grid<T, Axes ...>) is an iterable
+/// object of space points, such as a vector< ... >
+template <typename grid_t>
+class BinnedGroup {
+ public:
+#ifndef DOXYGEN
+  friend BinnedGroupIterator<grid_t>;
+#endif
```

Comment:
I think I figured out what the problem with this was before. I believe you can just do

```suggestion
  friend class BinnedGroupIterator<grid_t>;
```